src/output/help.rs

```diff
@@ -284,7 +282,13 @@ impl<'help, 'app, 'parser, 'writer> Help<'help, 'app, 'parser, 'writer> {
             }
 
             let sep = if arg.is_set(ArgSettings::RequireEquals) {
-                "="
+                if is_optional_val {
+                    "[="
+                } else {
+                    "="
+                }
+            } else if is_optional_val {
+                " ["
             } else {
                 " "
             };
```

Comment:
This function is not just for printing the long but for also printing the `=`, regardless of long or short, making it a misnomer.  We are making that worse by adding in the value.

At minimum, we should rename the function.

However, I suspect there is a better solution: we move the `=` to the `val`.

Reply:
Thanks for pointing this out. I'll rearrange so that the long, val, and alignment concerns are separated and that the separator and bracketing are part of the val concern.